Solve Home Screen Issues

M-Abdelnabi opened the pull request:
related to issues numbers ( 116 , 118 , 119 )

M-Abdelnabi committed:
Fix: Display correct year in Upcoming Media Card related to issue number #119

The `HomeScreen.kt` was updated to correctly display the year of release for upcoming media. Previously, it was attempting to display the entire `LocalDate` object as a string. Now, it correctly accesses the `year` property of the `yearOfRelease` field.

M-Abdelnabi committed:
Fix: Display correct year in MoodPickerDialog related to issue number #118

The `MoodPickerDialog.kt` was incorrectly displaying the `yearOfRelease` as a `LocalDate` object instead of the year itself. This has been corrected by accessing the `.year` property of the `yearOfRelease` object.

M-Abdelnabi committed:
Refactor: Adjust padding of the top of mood picker related to issue number #118

The bottom padding in the `LazyRow` within `HomeSection.kt` has been removed. The `contentPadding` has been changed from a uniform 16.dp to horizontal padding of 16.dp.

M-Abdelnabi committed:
Update text styles in MoodPicker related to issue number #118

The `MoodPicker` composable has been updated to use different text styles for its title and question elements:

- The title now uses `Theme.textStyle.label.medium` and its color is set to `Theme.colors.text.title`.
- The question text color is now explicitly set to `Theme.colors.text.body`.

M-Abdelnabi committed:
adjust fire icon size in top rating section related to issue number #116

The `HomeSection.kt` composable has been updated:
- The icon within the section header now has a fixed size of 16dp width and 18dp height.
- The top padding for the `SectionHeader` component has been removed.

M-Abdelnabi committed:
adjust padding from top of top rating section related to issue number #116

The top padding for the `TopRatingComponent` within the `HomeScreen` has been increased from `8.dp` to `24.dp`.

M-Abdelnabi committed:
adjust padding from top of continue watching section related to issue number #116

The top padding for the `TopRatingComponent` within the `HomeScreen` has been increased from `8.dp` to `24.dp`.

M-Abdelnabi committed:
Refactor: Update icon size beside popular title

The icon size in the `HomeSlider.kt` component has been explicitly set to `width = 16.dp` and `height = 18.dp`.

M-Abdelnabi committed:
Update TopAppBar title style in HomeScreen related to issue number #116

The `HomeScreen.kt` now explicitly sets the `titleTextStyle` for its `TopAppBar`. The `nicoMoji` font family is applied to the title, with specific font size, weight, line height, and letter spacing.

This change also involved updating the `TopAppBar.kt` component to:
- Accept a `titleTextStyle` parameter, defaulting to `Theme.textStyle.label.medium` if a subtitle exists, or `Theme.textStyle.title.large` otherwise.
- Apply the provided or default `titleTextStyle` to the title `Text` composable.
- Set a fixed `lineHeight` of `16.sp` for the subtitle `Text` composable.
- Removed a commented-out border modifier from the icon item within the `TopAppBar`.

M-Abdelnabi committed:
Update TopAppBar styling in HomeScreen related to issue number #116

The background color of the search icon in the `HomeScreen`'s `TopAppBar` has been changed from `Theme.colors.surfaceHigh` to `Theme.colors.primaryVariant`.

Additionally, in the `TopAppBar` component:
- The title `Text` now has a bottom padding of `2.dp`.
- The `Icon` within `IconItem` now has a fixed size of `24.dp`.

Joseph-Sameh-0 commented on feature/home/homeUi/src/main/java/com/feature/home/homeUi/screen/home/HomeScreen.kt:
```diff
@@ -254,6 +258,13 @@ fun HomeScreenContent(
     TopAppBar(
         title = stringResource(com.feature.home.homeUi.R.string.aflami),
         subtitle = "More than just watching.",
+        titleTextStyle = TextStyle(
+            fontSize = 14.sp,
+            fontFamily = nicoMoji,
+            fontWeight = FontWeight.Normal,
+            lineHeight = 20.sp,
+            letterSpacing = 0.25.sp
+        ),
         modifier = Modifier
```
don't add hardcoded text style
use the theme one from design system

M-Abdelnabi replied:
Doesn't exist in design system
This font and its style only used here

Joseph-Sameh-0 replied:
add it

Joseph-Sameh-0 commented:
update the release info

M-Abdelnabi committed:
feat: Add logoText style to AflamiTextStyle

A new `logoText` property of type `TextStyle` has been added to the `AflamiTextStyle` data class.

The `DefaultTextStyle` has been updated to include a definition for `logoText`, utilizing the `nicoMoji` font family with specific font size, weight, line height, and letter spacing.

M-Abdelnabi committed:
replace hard coded of text style to use design system

The `titleTextStyle` for the `TopAppBar` in `HomeScreen.kt` has been updated.
Instead of using a custom `TextStyle`, it now utilizes `Theme.textStyle.logoText`.

M-Abdelnabi committed:
Merge branch 'develop' into refactor/home-screen-issues

M-Abdelnabi committed:
Update app version and release notes

The `versionName` in `app/release-info.txt` has been updated from `0.3.4` to `0.3.6`.

The release notes have been changed from "- update exception handing in MoviesCategoriesRepository" to "- solve some issues of home screen".

M-Abdelnabi committed:
Merge branch 'develop' into refactor/home-screen-issues

M-Abdelnabi committed:
Merge branch 'develop' into refactor/home-screen-issues

# Conflicts:
#	app/release-info.txt
#	designsystem/src/main/java/com/paris_2/aflami/designsystem/components/TopAppBar.kt
#	feature/home/homeUi/src/main/java/com/feature/home/homeUi/screen/home/HomeScreen.kt
#	feature/home/homeUi/src/main/java/com/feature/home/homeUi/screen/home/components/HomeSlider.kt

M-Abdelnabi committed:
Refactor: Update import in HomeSlider.kt

The `size` import from `androidx.compose.foundation.layout` has been added to `HomeSlider.kt`.

M-Abdelnabi committed:
Update app version and release notes

The `versionName` in `app/release-info.txt` has been updated from `0.3.6` to `0.3.9`.
The release notes have been changed to "- solve some issues of home screen".

M-Abdelnabi committed:
Merge branch 'develop' into refactor/home-screen-issues

# Conflicts:
#	app/release-info.txt

M-Abdelnabi committed:
Refactor: Reorder IconBox parameters

The `tint` and `modifier` parameters in the `IconBox` composable function within `TopAppBar.kt` have been reordered. `modifier` now precedes `tint`.

M-Abdelnabi committed:
Merge branch 'develop' into refactor/home-screen-issues

# Conflicts:
#	app/release-info.txt
